nncf/experimental/torch/nas/bootstrapNAS/elasticity/elastic_width.py

```diff
@@ -672,35 +675,41 @@ def activate_subnet_for_config(self, config: ElasticWidthConfig) -> None:
             if not was_set and node_name not in names_of_processed_nodes:
                 nncf_logger.debug(f"input width was not set in scope={node.node_name}")
 
-            if self._add_dynamic_inputs:
-                if node_name in self._add_dynamic_inputs and not was_set:
-                    nncf_logger.debug(f"setting input width by user's request for scope={node_name}")
-                    nodes_to_check = [node]
-                    while any(elem is None for elem in input_masks):
-                        previous_nodes = []
-                        for node in nodes_to_check:
-                            previous_nodes.append(self._propagation_graph.get_previous_nodes(node))
-                        nodes_to_check.clear()
-                        previous_nodes = [item for nodes in previous_nodes for item in nodes]
-                        if not previous_nodes:
-                            break
-                        for previous in previous_nodes:
-                            if "output_mask" in previous.data:
-                                if previous.data["output_mask"] is not None:
-                                    input_masks.append(previous.data["output_mask"])
-                                    input_masks = [i for i in input_masks if i]
-                                else:
-                                    nodes_to_check.append(previous)
+    def get_input_masks_with_add_dynamic_inputs(self, node: NNCFNode, graph: NNCFGraph) -> List[Optional[NNCFTensor]]:
+        """
+        Returns input masks for all inputs of given NNCFNode.
+        :param node: Given NNCFNode.
+        :param graph: Graph to work with.
+        :return: Input masks.
+        """
+        retval = []
+        input_masks = [input_node.data["output_mask"] for input_node in graph.get_previous_nodes(node)]
+        if self._add_dynamic_inputs:
```

Comment:
Here is the place that `get_input_masks_with_add_dynamic_inputs` computes input masks of operators in `add_dynamic_inputs`.